src/diagonal.jl

```diff
@@ -332,6 +332,15 @@ function (*)(D::Diagonal, V::AbstractVector)
     return D.diag .* V
 end
 
+function (*)(A::AdjOrTransAbsMat, D::Diagonal)
+    adj = wrapperop(A)
+    copy(adj(adj(D) * adj(A)))
```

Comment:
It doesn't look like we have any optimized methods for `adjoint(diagonal) * matrix`, or am I missing something?

Reply:
`Adjoint` of a `Diagonal` produces a `Diagonal`